pgx-pg-sys/build.rs

```diff
@@ -793,16 +793,17 @@ fn apply_pg_guard(items: &Vec<syn::Item>) -> eyre::Result<proc_macro2::TokenStre
     for item in items {
         match item {
             Item::ForeignMod(block) => {
+                let abi = &block.abi;
                 for item in &block.items {
                     match item {
                         ForeignItem::Fn(func) => {
                             // Ignore other functions -- this will often be
                             // variadic functions that we can't safely wrap.
-                            if let Ok(tokens) = PgGuardRewriter::new().foreign_item_fn(func) {
+                            if let Ok(tokens) = PgGuardRewriter::new().foreign_item_fn(func, abi) {
                                 out.extend(tokens);
                             }
                         }
-                        other => out.extend(quote! { extern "C" { #other } }),
+                        other => out.extend(quote! { #abi { #other } }),
```

Comment:
Yes, it's entirely possible for a Unix system to have non-`extern "C"` bindings (albeit less common than on Windows... but with Postgres we're deep into the land of doing things that are less common, e.g. we mess with `sigsetjmp` at all). It also makes the changes forward-resilient against any changes necessary for e.g. `extern "C-unwind"`.

Reply:
> with Postgres we're deep into the land of doing things that are less common

😭

> It also makes the changes forward-resilient against any changes necessary for e.g. extern "C-unwind".

Oh, support for C-unwind isn't even a case I thought about! Good point!